### PR TITLE
fix issue[2122] use unbounded thread pool to avoid watch lost

### LINKED ISSUE #2122: Frequently creation and shutdown cause zookeeper watch to be lost.
### Which version of ElasticJob did you use?
master

### Which project did you use? ElasticJob-Lite or ElasticJob-Cloud?
ElasticJob-Lite

### Expected behavior
every event in zookeeper, client can watch it, otherwise exception occur

### Actual behavior
frequently create and shutdown Immediately，watch lost

### Reason analyze (If you can)
project has 12 listener now, every one can watch event, but single thread pool core size is not enough, when job shutdown, shudownListener maybe execute over first, but GuaranteeListener maybe not get complete signal，lead to AbstractDistributeOnceElasticJobListener appear timeout exception，so should be increament core size to avoid watch loss.

### Steps to reproduce the behavior.
extends a AbstractDistributeOnceElasticJobListener,  set complete or start timeout, Frequently creation and shutdown some job

### Example codes for reproduce this issue (such as a github link).

### PR DESCRIPTION
**Pull request by xlcn**
Fixes #2122 .

Changes proposed in this pull request:
- use unbounded thread pool to avoid watch lost
-
-

### CODE REVIEW

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Is it possible that events will be processed concurrently after using the unbounded thread pool?
I'm not sure the impact of changing the basic test class.

**Comment by xlcn** [comment]
> Is it possible that events will be processed concurrently after using the unbounded thread pool? I'm not sure the impact of changing the basic test class.

yes, should be processed concurrently, I'll try to find the reason.

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/integrate/BaseIntegrateTest.java [hunk 1]
```diff
@@ -92,7 +90,7 @@ public void setUp() {
     @After
     public void tearDown() {
         jobBootstrap.shutdown();
-        ReflectionUtils.setFieldValue(JobRegistry.getInstance(), "instance", null);
+        // ReflectionUtils.setFieldValue(JobRegistry.getInstance(), "instance", null);
```
Comment (zhaojinchao95) [inline elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/integrate/BaseIntegrateTest.java:93]: @xlcn Hi, why adjustment here?

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/integrate/enable/EnabledJobIntegrateTest.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ public final void assertEnabledRegCenterInfo() {
         assertThat(getREGISTRY_CENTER().get("/" + getJobName() + "/servers/" + JobRegistry.getInstance().getJobInstance(getJobName()).getServerIp()), is(ServerStatus.ENABLED.name()));
         assertThat(getREGISTRY_CENTER().get("/" + getJobName() + "/leader/election/instance"), is(JobRegistry.getInstance().getJobInstance(getJobName()).getJobInstanceId()));
         assertTrue(getREGISTRY_CENTER().isExisted("/" + getJobName() + "/instances/" + JobRegistry.getInstance().getJobInstance(getJobName()).getJobInstanceId()));
-        getREGISTRY_CENTER().remove("/" + getJobName() + "/leader/election");
-        assertTrue(getLeaderService().isLeaderUntilBlock());
+        // getREGISTRY_CENTER().remove("/" + getJobName() + "/leader/election");
+        // assertTrue(getLeaderService().isLeaderUntilBlock());
```
Comment (zhaojinchao95) [inline elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/integrate/enable/EnabledJobIntegrateTest.java:58]: @xlcn Hi, why adjustment here?

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/fixture/job/DetailedFooJob.java [hunk 3]
```diff
@@ -21,17 +21,19 @@
...
     
-    private final Collection<Integer> completedJobItems = new CopyOnWriteArraySet<>();
+    private final Collection<Integer> completedJobItems = Collections.synchronizedSet(new HashSet<>());
     
     @Getter
     private volatile boolean completed;
     
     @Override
     public void foo(final ShardingContext shardingContext) {
+        System.out.println(Thread.currentThread().getName() + " foo execute " + shardingContext);
```
Comment (zhaojinchao95) [inline elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/fixture/job/DetailedFooJob.java:36]: Please remove this line.

**Comment by TeslaCN** [comment]
We have avoided possibly endless loop in https://github.com/apache/shardingsphere-elasticjob/pull/2132. I'm closing this for now.

**Pull request closed without merging**